substrate/client/tracing/src/block/mod.rs

```diff
@@ -228,7 +231,21 @@ where
 			if let Err(e) = dispatcher::with_default(&dispatch, || {
 				let span = tracing::info_span!(target: TRACE_TARGET, "trace_block");
 				let _enter = span.enter();
-				self.client.runtime_api().execute_block(parent_hash, block)
+
+				let mut runtime_api = self.client.runtime_api();
+
+				// Enable proof recording if requested
+				if self.record_proof {
```

Comment:
You can just always enable proof recording. If the extension is set and not used, there is no downside.

Reply:
I see a downside for solochains that use Substrate: if they use a pallet or a library that calls the host function `get_proof_size`, it will create a discrepancy between the actual block execution and the tracing.

Reply:
The host function is not part of default Substrate. You get this only by using cumulus stuff. For sure, someone could still do this.

Maybe a better way would be to provide a way to override the block execution. This way someone could enable the proof generation if required. I would also have some future use case for this.